File: sdks/dotnet/sulfone-helium/Api/Template/Lambda.cs
using sulfone_helium.Domain.Core;

namespace sulfone_helium.Api.Template;

public class LambdaTemplate(Func<IInquirer, IDeterminism, Task<Cyan>> f) : ICyanTemplate
{
    public Task<Cyan> Template(IInquirer inquirer, IDeterminism determinism)
    {
        return f(inquirer, determinism);
    }
}

File: sdks/node/src/api/template/lambda.ts
import type { IInquirer } from "../../domain/core/inquirer.js";
import type { IDeterminism } from "../../domain/core/deterministic.js";
import type { Cyan } from "../../domain/core/cyan.js";
import type { ICyanTemplate } from "../../domain/core/cyan_script.js";

type LambdaTemplateFn = (
  inquirer: IInquirer,
  determinism: IDeterminism,
) => Promise<Cyan>;

class LambdaTemplate implements ICyanTemplate {
  private readonly _f: LambdaTemplateFn;

  constructor(f: LambdaTemplateFn) {
    this._f = f;
  }

  template(inquirer: IInquirer, determinism: IDeterminism): Promise<Cyan> {
    return this._f(inquirer, determinism);
  }
}

export type { LambdaTemplateFn };
export { LambdaTemplate };
